src/MongoDB.Analyzer/Core/TypesProcessor.cs

```diff
@@ -248,25 +283,24 @@ private TypeSyntax CreateTypeSyntaxForSymbol(ITypeSymbol typeSymbol)
             namedTypeSymbol.TypeArguments.Length == 1 &&
             namedTypeSymbol.IsSupportedCollection())
         {
-            var underlyingTypeSyntax = CreateTypeSyntaxForSymbol(namedTypeSymbol.TypeArguments.Single());
-            var listSyntax = SyntaxFactory.GenericName(
-                SyntaxFactory.Identifier("List"),
-                SyntaxFactory.TypeArgumentList(SyntaxFactory.SeparatedList(new[] { underlyingTypeSyntax })));
+            var underlyingTypeSyntaxes = namedTypeSymbol.TypeArguments.Select(typeSymbol => CreateTypeSyntaxForSymbol(typeSymbol));
```

Comment:
We are checking: `namedTypeSymbol.TypeArguments.Length == 1`, so looks like we can't have multiple syntaxes here.

Reply:
Done